### PR TITLE
[k2] implement simple HTTP server

### PR DESCRIPTION
**Pull request by apolyakov**
In this PR I add:

- almost complete initialization of HTTP superglobals
- simplified implementation of header builtin

### CODE REVIEW

File: compiler/make/objs-to-k2-component-target.h [hunk 1]
```diff
@@ -59,7 +59,9 @@ class Objs2K2ComponentTarget : public Target {
     }
     // add vendored statically linking libs
     std::vector<std::string> libs = split(RUNTIME_LINK_LIBS);
+    ss << " -Wl,--start-group ";
     std::for_each(libs.cbegin(), libs.cend(), [&ss](const auto &lib) noexcept { ss << lib << " "; });
+    ss << " -Wl,--end-group ";
```
Comment (mkornaukhov03) [inline compiler/make/objs-to-k2-component-target.h:64]: It isn't mac os compatible.
See `objs-to-bin-target.h`

File: runtime-common/stdlib/server/url-functions.cpp [hunk 2]
```diff
@@ -471,3 +496,38 @@ mixed f$parse_url(const string &s, int64_t component) noexcept {
...
+  arr.set_value(key, value);
+}
+
+void f$parse_str(const string &str, mixed &arr) noexcept {
+  if (f$trim(str).empty()) {
+    arr = array<mixed>();
+    return;
+  }
+
+  arr = array<mixed>();
```
Comment (mkornaukhov03) [inline runtime-common/stdlib/server/url-functions.cpp:517]: `arr = array<mixed>();` unconditionally

File: runtime/url.cpp [hunk 3]
```diff
@@ -4,80 +4,17 @@
...
 //
-//  if (result_len < 0) {
-//    return string();
-//  }
+//   if (result_len < 0) {
+//     return string();
+//   }
 //
-//  res.shrink (result_len);
-//  return res;
```
Comment (mkornaukhov03) [inline runtime/url.cpp:25]: extra changes

File: runtime-common/stdlib/server/url-functions.cpp (hunk 2 above)
Comment (apolyakov) [reply runtime-common/stdlib/server/url-functions.cpp:517]: It's a code from `runtime`. That functionality is just moved into `runtime-common`, I don't want to change it now.

File: runtime/url.cpp (hunk 3 above)
Comment (apolyakov) [reply runtime/url.cpp:25]: Same as previous note. I don't want to change it in a PR with another goal.

Comment (mkornaukhov03) [reply runtime/url.cpp:25]: That's the point, but you do change these lines

File: runtime-light/stdlib/server/http-functions.cpp [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+#include <cstdint>
+#include <ranges>
+#include <string_view>
+#include <utility>
+
+#include "runtime-common/core/utils/kphp-assert-core.h"
+#include "runtime-light/server/http/http-server-state.h"
+
+#include "absl/strings/ascii.h"
+#include "absl/strings/str_split.h"
```
[the remark this answers is not in the document]
Comment (apolyakov) [reply runtime-light/stdlib/server/http-functions.cpp:18]: Agree. Refactored it

File: runtime/url.cpp (hunk 3 above)
Comment (apolyakov) [reply runtime/url.cpp]: Sorry, misunderstanding. Fixed

**Pull request merged**